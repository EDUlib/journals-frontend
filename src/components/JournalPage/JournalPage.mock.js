
const mockJournalPage = {
  title: 'dummy title',
  subTitle: 'dummy subtitle',
  getPage: jest.fn(),
  setPageVisit: jest.fn(),
  getPreview: jest.fn(),
  is_preview: false,
  startedFetching: true,
  finishedFetching: true,
  breadCrumbs: [{ id: 1, title: 'dummy' }],
  displayLastPublishedDate: false,
  pageId: 1,
  userId: 2,
  nextPage: 3,
  previousPage: 1,
  match: { params: { journalAboutId: '12', pageId: '1', previewId: '13' }, url: 'dummy.com/dummy' },
  body: [],
};

export default mockJournalPage;
