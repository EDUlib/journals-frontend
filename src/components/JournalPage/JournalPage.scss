h1.article-title{
  margin-top: 25px;
  margin-bottom: 10px;
}
h2.subtitle {
  font-size: 20px;
  line-height: 1.5;
  margin: 0px;
}
div.date-and-author{
  margin-top: 10px;

  p.muted-text{
    margin: 0px;
  }
}
div.journal-page-body{
  margin-top: 30px;
  margin-bottom: 30px;
}
.body-element{
  margin-top: 30px;
  margin-bottom: 30px;
}
