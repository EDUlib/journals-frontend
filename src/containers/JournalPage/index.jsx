import { connect } from 'react-redux';

import Page from '../../components/JournalPage';

const mapStateToProps = () => (
  {
  }
);

const mapDispatchToProps = () => (
  {
  }
);

const JournalPage = connect(
  mapStateToProps,
  mapDispatchToProps,
)(Page);

export default JournalPage;
