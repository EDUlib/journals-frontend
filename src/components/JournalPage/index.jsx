import React from 'react';
import Moment from 'react-moment';
import PropTypes from 'prop-types';
import {
  RICH_TEXT,
  RAW_HTML,
  PDF,
  XBLOCK_VIDEO,
  IMAGE,
} from '../../data/constants/contentTypes';
import UnauthorizedPage from '../UnauthorizedPage';

import RawHTMLViewer from '../RawHTMLViewer';
import ImageViewerContainer from '../../containers/ImageViewerContainer';
import PageNavigationButtons from '../PageNavigationButtons';
import PDFViewerContainer from '../../containers/PDFViewerContainer';
import ViewerBorder from '../ViewerBorder';
import BreadCrumbs from '../BreadCrumbs';
import XBlockVideoViewer from '../XBlockVideoViewer';

import './JournalPage.scss';


class JournalPage extends React.Component {
  componentDidMount() {
    this.props.getPage(this.props.match.params.pageId);
  }
  componentDidUpdate(prevProps) {
    // If we try to change Journal pages
    if (prevProps.match.params.pageId !== this.props.match.params.pageId) {
      this.props.getPage(this.props.match.params.pageId);
    }
    // If we successfully change Journal pages
    if (prevProps.pageId !== this.props.pageId) {
      this.trackVisit();
    }
  }

  trackVisit() {
    if (this.props.userId != null) {
      this.props.setPageVisit(this.props.userId, this.props.match.params.pageId);
    }
  }

  render() {
    const baseUrl = !this.props.is_preview ? `/${this.props.match.params.journalAboutId}/pages` : '';
    const previousPageUrl = this.props.previousPage ? `${baseUrl}/${this.props.previousPage}` : '';
    const nextPageUrl = this.props.nextPage ? `${baseUrl}/${this.props.nextPage}` : '';
    return (
      this.props.fetchPageSuccess ? (
        <UnauthorizedPage />
      ) : (
        <div className="page">
          { this.props.breadCrumbs.length > 0 &&
            <BreadCrumbs
              ancestorPages={this.props.breadCrumbs}
              journalId={this.props.match.params.journalAboutId}
            />
          }
          <h1 className="article-title">{this.props.title}</h1>
          <h2 className="subtitle">{this.props.subTitle}</h2>
          <div className="date-and-author">
            {
              this.props.displayLastPublishedDate &&
              <p className="muted-text">
                <Moment date={this.props.lastPublishedDate} format="MMMM DD, YYYY" />
              </p>
            }
            <p className="muted-text">{ this.props.author.trim() && `By ${this.props.author}` }</p>
          </div>
          <div className="journal-page-body">
            {
              this.props.body.map((el) => {
                switch (el.type) {
                  case RICH_TEXT:
                  case RAW_HTML:
                    // rich text and raw HTML use the same component
                    return (
                      <div className="body-element">
                        <RawHTMLViewer key={el.id} content={el.value} />
                      </div>
                    );
                  case PDF:
                    return (
                      <div className="body-element">
                        <ViewerBorder
                          key={el.id}
                          spanId={el.value.span_id}
                          title={el.value.title}
                          spanFullWidth
                        >
                          <PDFViewerContainer url={el.value.url} title={el.value.title} />
                        </ViewerBorder>
                      </div>
                    );
                  case XBLOCK_VIDEO:
                    return (
                      <div className="body-element">
                        <ViewerBorder
                          spanId={el.value.span_id}
                          title={el.value.title}
                          spanFullWidth
                        >
                          <XBlockVideoViewer url={el.value.view_url} title={el.value.title} />
                        </ViewerBorder>
                      </div>
                    );
                  case IMAGE:
                    return (
                      <div className="body-element">
                        <ViewerBorder
                          key={el.id}
                          spanId={el.value.span_id}
                          title={el.value.title}
                          caption={el.value.caption}
                        >
                          <ImageViewerContainer url={el.value.url} altText={el.value.title} />
                        </ViewerBorder>
                      </div>
                    );
                  default:
                    return <div key={el.id}>No matching component</div>;
                }
              })
            }
          </div>
          {
            !this.props.is_preview &&
              <PageNavigationButtons prev={previousPageUrl} next={nextPageUrl} />
          }
        </div>
      )
    );
  }
}

JournalPage.defaultProps = {
  title: '',
  subTitle: '',
  displayLastPublishedDate: false,
  lastPublishedDate: '',
  author: '',
  breadCrumbs: [],
  body: [],
  getPage: () => {},
  setPageVisit: () => {},
  fetchPageSuccess: false,
  nextPage: null,
  previousPage: null,
  is_preview: false,
  pageId: 0,
  userId: null,
};

JournalPage.propTypes = {
  title: PropTypes.string,
  subTitle: PropTypes.string,
  displayLastPublishedDate: PropTypes.bool,
  lastPublishedDate: PropTypes.string,
  author: PropTypes.string,
  breadCrumbs: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
  })),
  body: PropTypes.arrayOf(PropTypes.object),
  getPage: PropTypes.func,
  setPageVisit: PropTypes.func,
  match: PropTypes.shape({
    params: PropTypes.shape({
      pageId: PropTypes.string,
      journalAboutId: PropTypes.string,
    }),
    url: PropTypes.string,
  }).isRequired,
  fetchPageSuccess: PropTypes.bool,
  nextPage: PropTypes.number,
  previousPage: PropTypes.number,
  is_preview: PropTypes.bool,
  userId: PropTypes.number,
  pageId: PropTypes.number,
};


export default JournalPage;
