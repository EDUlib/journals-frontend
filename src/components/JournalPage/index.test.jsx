import React from 'react';
import { mount } from 'enzyme';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';

import JournalPage from './index';
import mockJournalPage from './JournalPage.mock';
import {
  RICH_TEXT,
  RAW_HTML,
  PDF,
  XBLOCK_VIDEO,
  IMAGE,
} from '../../data/constants/contentTypes';


const mockStore = configureMockStore([thunk]);
const store = mockStore({ siteInfo: { serverBaseUrl: 'example.com' } });
const JournalPageWrapper = props => (
  <MemoryRouter>
    <Provider store={store}>
      <JournalPage
        {...mockJournalPage}
        {...props}
      />
    </Provider>
  </MemoryRouter>
);

describe('<JournalPage />', () => {
  describe('renders correctly', () => {
    it('with RAW HTML component', () => {
      const body = [
        { type: RAW_HTML, id: 1, value: 'body with raw_html' },
      ];
      const tree = renderer
        .create((
          <JournalPageWrapper
            body={body}
          />
        )).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('with RICH TEXT component', () => {
      const body = [
        { type: RICH_TEXT, id: 2, value: 'body with rich_text' },
      ];
      const tree = renderer
        .create((
          <JournalPageWrapper
            body={body}
          />
        )).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('with PDF component', () => {
      const body = [
        { type: PDF, id: 3, value: { span_id: 'p-id', title: 'p-title', url: 'example.com/PDF' } },
      ];
      const tree = renderer
        .create((
          <JournalPageWrapper
            body={body}
          />
        )).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('with IMAGE component', () => {
      const body = [
        {
          type: IMAGE,
          id: 5,
          value: {
            span_id: 'i-id', title: 'i-title', url: 'ex.pk/image', caption: 'dummy',
          },
        },
      ];
      const tree = renderer
        .create((
          <JournalPageWrapper
            body={body}
          />
        )).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('with XBLOCK VIDEO component', () => {
      const body = [
        { type: XBLOCK_VIDEO, id: 4, value: { span_id: 'v-id', title: 'v-title', view_url: 'ex.pk/video' } },
      ];
      const tree = renderer
        .create((
          <JournalPageWrapper
            body={body}
          />
        )).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('with DUMMY component', () => {
      const body = [
        { type: 'dummy', id: 1 },
      ];
      const tree = renderer
        .create((
          <JournalPageWrapper
            body={body}
          />
        )).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('with last publish date.', () => {
      const mockedGetPage = jest.fn();
      const displayLastPublishedDate = true;
      const lastPublishedDate = '1976-04-19';
      const author = 'jon doe';
      const tree = renderer
        .create((
          <JournalPageWrapper
            displayLastPublishedDate={displayLastPublishedDate}
            lastPublishedDate={lastPublishedDate}
            author={author}
            getPage={mockedGetPage}
          />
        )).toJSON();
      expect(mockedGetPage).toHaveBeenCalledWith(mockJournalPage.match.params.pageId);
      expect(tree).toMatchSnapshot();
    });

    it('error page with 404 status code.', () => {
      const error = { response: { status: 404 }, message: '' };
      const tree = renderer
        .create((
          <JournalPageWrapper
            error={error}
          />
        ));
      expect(tree).toMatchSnapshot();
    });

    it('error page with 403 status code.', () => {
      const error = { response: { status: 403 }, message: '' };
      const tree = renderer
        .create((
          <JournalPageWrapper
            error={error}
          />
        ));
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Data fetching', () => {
    it('not finished and checks the loading page.', () => {
      const finishedFetching = false;
      const tree = renderer
        .create((
          <JournalPageWrapper
            finishedFetching={finishedFetching}
          />
        ));
      expect(tree).toMatchSnapshot();
    });

    it('not finished and test the trackVisit function.', () => {
      const tree = mount((
        <JournalPageWrapper />
      ));
      // testing "componentDidUpdate" function
      const newProps = Object.assign({}, mockJournalPage);
      const mockedSetPageVisit = jest.fn();

      // mocked the 'ScrollIntoView' function and adding a div in body.
      const mockedScrollIntoView = jest.fn();
      window.HTMLElement.prototype.scrollIntoView = mockedScrollIntoView;
      document.body.innerHTML = '<div class="header"></div>';

      newProps.pageId = 12;
      newProps.match.params.pageId = '2';
      newProps.setPageVisit = mockedSetPageVisit;

      tree.setProps(newProps);
      expect(mockedScrollIntoView).toHaveBeenCalled();
      expect(mockedSetPageVisit).toHaveBeenCalledWith(
        newProps.userId,
        newProps.match.params.pageId,
        newProps.match.params.journalAboutId,
      );
    });
  });
});
