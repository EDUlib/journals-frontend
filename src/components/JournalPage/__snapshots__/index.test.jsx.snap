// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JournalPage /> Data fetching not finished and checks the loading page. 1`] = `
<div
  className="journal-page-body"
>
  Loading...
</div>
`;

exports[`<JournalPage /> renders correctly error page with 403 status code. 1`] = `
<div>
  <div>
    <div
      className="row mt-4"
    >
      <div
        className="col"
      >
        <h1>
          Error
        </h1>
        <div
          className="alert fade alert-danger show"
          hidden={false}
          role="alert"
        >
          <div
            className="alert-dialog"
          >
            <div
              className="alert-dialog"
            >
              <div
                className="icon mr-2"
              >
                <span
                  aria-hidden={true}
                  className="fa fa-times-circle"
                  id="Icon1"
                />
              </div>
              <div
                className="message"
              >
                <span>
                  An unknown error has occured.
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<JournalPage /> renders correctly error page with 404 status code. 1`] = `
<div>
  <div
    className="mt-3"
  >
    <div
      className="text-center py-5"
    >
      <h1>
        404
      </h1>
      <p
        className="lead"
      >
        Oops, sorry we can't find that page!
      </p>
      <p>
        Either something went wrong or the page doesn't exist anymore.
      </p>
    </div>
  </div>
</div>
`;

exports[`<JournalPage /> renders correctly with DUMMY component 1`] = `
<div
  className="page"
>
  <ul
    className="bread-crumb-list"
  >
    <li>
      <a
        className="bread-crumb-link"
        href="/12/pages/1"
        onClick={[Function]}
      >
        <u>
          dummy
        </u>
      </a>
      <span
        aria-hidden={true}
        className="fa fa-angle-right bread-crumb-arrow"
        id="Icon1"
      />
    </li>
  </ul>
  <h1
    className="article-title"
  >
    dummy title
  </h1>
  <h2
    className="subtitle"
  >
    dummy subtitle
  </h2>
  <div
    className="date-and-author"
  >
    <p
      className="muted-text"
    >
      
    </p>
  </div>
  <div
    className="journal-page-body"
  >
    <div>
      No matching component
    </div>
  </div>
  <div
    className="page-nav-btns"
  >
    <a
      className="btn btn-outline-secondary nav-btn"
      href="/12/pages/1"
      onClick={[Function]}
    >
      <div
        className="btn-label"
      >
        <span
          aria-hidden={true}
          className="fa fa-angle-left"
          id="Icon1"
        />
        <span
          className="btn-text"
        >
          Previous
        </span>
      </div>
    </a>
    <a
      className="btn btn-outline-secondary nav-btn"
      href="/12/pages/3"
      onClick={[Function]}
    >
      <div
        className="btn-label"
      >
        <span
          className="btn-text"
        >
          Next
        </span>
        <span
          aria-hidden={true}
          className="fa fa-angle-right"
          id="Icon1"
        />
      </div>
    </a>
  </div>
</div>
`;

exports[`<JournalPage /> renders correctly with IMAGE component 1`] = `
<div
  className="page"
>
  <ul
    className="bread-crumb-list"
  >
    <li>
      <a
        className="bread-crumb-link"
        href="/12/pages/1"
        onClick={[Function]}
      >
        <u>
          dummy
        </u>
      </a>
      <span
        aria-hidden={true}
        className="fa fa-angle-right bread-crumb-arrow"
        id="Icon1"
      />
    </li>
  </ul>
  <h1
    className="article-title"
  >
    dummy title
  </h1>
  <h2
    className="subtitle"
  >
    dummy subtitle
  </h2>
  <div
    className="date-and-author"
  >
    <p
      className="muted-text"
    >
      
    </p>
  </div>
  <div
    className="journal-page-body"
  >
    <div
      className="body-element"
    >
      <span
        id="i-id"
      >
        <div
          className="viewer-border"
        >
          <div
            className="grow"
          >
            <div
              className="viewer-title"
            >
              i-title
            </div>
          </div>
          <div
            className="viewer-content"
          >
            <img
              alt="i-title"
              className="image-block"
              src="ex.pk/image"
            />
          </div>
          <div
            className="viewer-caption grow"
          >
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "dummy",
                }
              }
            />
          </div>
        </div>
      </span>
    </div>
  </div>
  <div
    className="page-nav-btns"
  >
    <a
      className="btn btn-outline-secondary nav-btn"
      href="/12/pages/1"
      onClick={[Function]}
    >
      <div
        className="btn-label"
      >
        <span
          aria-hidden={true}
          className="fa fa-angle-left"
          id="Icon1"
        />
        <span
          className="btn-text"
        >
          Previous
        </span>
      </div>
    </a>
    <a
      className="btn btn-outline-secondary nav-btn"
      href="/12/pages/3"
      onClick={[Function]}
    >
      <div
        className="btn-label"
      >
        <span
          className="btn-text"
        >
          Next
        </span>
        <span
          aria-hidden={true}
          className="fa fa-angle-right"
          id="Icon1"
        />
      </div>
    </a>
  </div>
</div>
`;

exports[`<JournalPage /> renders correctly with PDF component 1`] = `
<div
  className="page"
>
  <ul
    className="bread-crumb-list"
  >
    <li>
      <a
        className="bread-crumb-link"
        href="/12/pages/1"
        onClick={[Function]}
      >
        <u>
          dummy
        </u>
      </a>
      <span
        aria-hidden={true}
        className="fa fa-angle-right bread-crumb-arrow"
        id="Icon1"
      />
    </li>
  </ul>
  <h1
    className="article-title"
  >
    dummy title
  </h1>
  <h2
    className="subtitle"
  >
    dummy subtitle
  </h2>
  <div
    className="date-and-author"
  >
    <p
      className="muted-text"
    >
      
    </p>
  </div>
  <div
    className="journal-page-body"
  >
    <div
      className="body-element"
    >
      <span
        id="p-id"
      >
        <div
          className="viewer-border w-100"
        >
          <div
            className="grow"
          >
            <div
              className="viewer-title"
            >
              p-title
              <button
                className="btn control-btn float-right"
                onBlur={[Function]}
                onClick={[Function]}
                onKeyDown={[Function]}
                type="button"
              >
                <span
                  aria-hidden={true}
                  className="fa fa-expand"
                  id="Icon1"
                />
              </button>
            </div>
          </div>
          <div
            className="fullscreen"
            style={undefined}
          >
            <div
              className="viewer-content"
            >
              <div
                className="embed-responsive responsive-pdf"
              >
                <iframe
                  className="embed-responsive-item"
                  src="example.com/static/pdf_js/web/viewer.html?file=example.com/PDF"
                  title="p-title"
                />
              </div>
            </div>
          </div>
          
        </div>
      </span>
    </div>
  </div>
  <div
    className="page-nav-btns"
  >
    <a
      className="btn btn-outline-secondary nav-btn"
      href="/12/pages/1"
      onClick={[Function]}
    >
      <div
        className="btn-label"
      >
        <span
          aria-hidden={true}
          className="fa fa-angle-left"
          id="Icon1"
        />
        <span
          className="btn-text"
        >
          Previous
        </span>
      </div>
    </a>
    <a
      className="btn btn-outline-secondary nav-btn"
      href="/12/pages/3"
      onClick={[Function]}
    >
      <div
        className="btn-label"
      >
        <span
          className="btn-text"
        >
          Next
        </span>
        <span
          aria-hidden={true}
          className="fa fa-angle-right"
          id="Icon1"
        />
      </div>
    </a>
  </div>
</div>
`;

exports[`<JournalPage /> renders correctly with RAW HTML component 1`] = `
<div
  className="page"
>
  <ul
    className="bread-crumb-list"
  >
    <li>
      <a
        className="bread-crumb-link"
        href="/12/pages/1"
        onClick={[Function]}
      >
        <u>
          dummy
        </u>
      </a>
      <span
        aria-hidden={true}
        className="fa fa-angle-right bread-crumb-arrow"
        id="Icon1"
      />
    </li>
  </ul>
  <h1
    className="article-title"
  >
    dummy title
  </h1>
  <h2
    className="subtitle"
  >
    dummy subtitle
  </h2>
  <div
    className="date-and-author"
  >
    <p
      className="muted-text"
    >
      
    </p>
  </div>
  <div
    className="journal-page-body"
  >
    <div
      className="body-element"
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "body with raw_html",
          }
        }
      />
    </div>
  </div>
  <div
    className="page-nav-btns"
  >
    <a
      className="btn btn-outline-secondary nav-btn"
      href="/12/pages/1"
      onClick={[Function]}
    >
      <div
        className="btn-label"
      >
        <span
          aria-hidden={true}
          className="fa fa-angle-left"
          id="Icon1"
        />
        <span
          className="btn-text"
        >
          Previous
        </span>
      </div>
    </a>
    <a
      className="btn btn-outline-secondary nav-btn"
      href="/12/pages/3"
      onClick={[Function]}
    >
      <div
        className="btn-label"
      >
        <span
          className="btn-text"
        >
          Next
        </span>
        <span
          aria-hidden={true}
          className="fa fa-angle-right"
          id="Icon1"
        />
      </div>
    </a>
  </div>
</div>
`;

exports[`<JournalPage /> renders correctly with RICH TEXT component 1`] = `
<div
  className="page"
>
  <ul
    className="bread-crumb-list"
  >
    <li>
      <a
        className="bread-crumb-link"
        href="/12/pages/1"
        onClick={[Function]}
      >
        <u>
          dummy
        </u>
      </a>
      <span
        aria-hidden={true}
        className="fa fa-angle-right bread-crumb-arrow"
        id="Icon1"
      />
    </li>
  </ul>
  <h1
    className="article-title"
  >
    dummy title
  </h1>
  <h2
    className="subtitle"
  >
    dummy subtitle
  </h2>
  <div
    className="date-and-author"
  >
    <p
      className="muted-text"
    >
      
    </p>
  </div>
  <div
    className="journal-page-body"
  >
    <div
      className="body-element"
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "body with rich_text",
          }
        }
      />
    </div>
  </div>
  <div
    className="page-nav-btns"
  >
    <a
      className="btn btn-outline-secondary nav-btn"
      href="/12/pages/1"
      onClick={[Function]}
    >
      <div
        className="btn-label"
      >
        <span
          aria-hidden={true}
          className="fa fa-angle-left"
          id="Icon1"
        />
        <span
          className="btn-text"
        >
          Previous
        </span>
      </div>
    </a>
    <a
      className="btn btn-outline-secondary nav-btn"
      href="/12/pages/3"
      onClick={[Function]}
    >
      <div
        className="btn-label"
      >
        <span
          className="btn-text"
        >
          Next
        </span>
        <span
          aria-hidden={true}
          className="fa fa-angle-right"
          id="Icon1"
        />
      </div>
    </a>
  </div>
</div>
`;

exports[`<JournalPage /> renders correctly with XBLOCK VIDEO component 1`] = `
<div
  className="page"
>
  <ul
    className="bread-crumb-list"
  >
    <li>
      <a
        className="bread-crumb-link"
        href="/12/pages/1"
        onClick={[Function]}
      >
        <u>
          dummy
        </u>
      </a>
      <span
        aria-hidden={true}
        className="fa fa-angle-right bread-crumb-arrow"
        id="Icon1"
      />
    </li>
  </ul>
  <h1
    className="article-title"
  >
    dummy title
  </h1>
  <h2
    className="subtitle"
  >
    dummy subtitle
  </h2>
  <div
    className="date-and-author"
  >
    <p
      className="muted-text"
    >
      
    </p>
  </div>
  <div
    className="journal-page-body"
  >
    <div
      className="body-element"
    >
      <span
        id="v-id"
      >
        <div
          className="viewer-border w-100"
        >
          <div
            className="grow"
          >
            <div
              className="viewer-title"
            >
              v-title
              <button
                className="btn control-btn float-right"
                onBlur={[Function]}
                onClick={[Function]}
                onKeyDown={[Function]}
                type="button"
              >
                <span
                  aria-hidden={true}
                  className="fa fa-expand"
                  id="Icon1"
                />
              </button>
            </div>
          </div>
          <div
            className="fullscreen"
            style={undefined}
          >
            <div
              className="viewer-content"
            >
              <div
                className="xblock-video-container embed-responsive responsive-video"
              >
                <iframe
                  allowFullScreen={true}
                  className="imported-xblock-video embed-responsive-item"
                  src="ex.pk/video"
                  title="v-title"
                />
              </div>
            </div>
          </div>
          
        </div>
      </span>
    </div>
  </div>
  <div
    className="page-nav-btns"
  >
    <a
      className="btn btn-outline-secondary nav-btn"
      href="/12/pages/1"
      onClick={[Function]}
    >
      <div
        className="btn-label"
      >
        <span
          aria-hidden={true}
          className="fa fa-angle-left"
          id="Icon1"
        />
        <span
          className="btn-text"
        >
          Previous
        </span>
      </div>
    </a>
    <a
      className="btn btn-outline-secondary nav-btn"
      href="/12/pages/3"
      onClick={[Function]}
    >
      <div
        className="btn-label"
      >
        <span
          className="btn-text"
        >
          Next
        </span>
        <span
          aria-hidden={true}
          className="fa fa-angle-right"
          id="Icon1"
        />
      </div>
    </a>
  </div>
</div>
`;

exports[`<JournalPage /> renders correctly with last publish date. 1`] = `
<div
  className="page"
>
  <ul
    className="bread-crumb-list"
  >
    <li>
      <a
        className="bread-crumb-link"
        href="/12/pages/1"
        onClick={[Function]}
      >
        <u>
          dummy
        </u>
      </a>
      <span
        aria-hidden={true}
        className="fa fa-angle-right bread-crumb-arrow"
        id="Icon1"
      />
    </li>
  </ul>
  <h1
    className="article-title"
  >
    dummy title
  </h1>
  <h2
    className="subtitle"
  >
    dummy subtitle
  </h2>
  <div
    className="date-and-author"
  >
    <p
      className="muted-text"
    >
      <time
        dateTime={"1976-04-19T00:00:00.000Z"}
      >
        April 19, 1976
      </time>
    </p>
    <p
      className="muted-text"
    >
      By jon doe
    </p>
  </div>
  <div
    className="journal-page-body"
  />
  <div
    className="page-nav-btns"
  >
    <a
      className="btn btn-outline-secondary nav-btn"
      href="/12/pages/1"
      onClick={[Function]}
    >
      <div
        className="btn-label"
      >
        <span
          aria-hidden={true}
          className="fa fa-angle-left"
          id="Icon1"
        />
        <span
          className="btn-text"
        >
          Previous
        </span>
      </div>
    </a>
    <a
      className="btn btn-outline-secondary nav-btn"
      href="/12/pages/3"
      onClick={[Function]}
    >
      <div
        className="btn-label"
      >
        <span
          className="btn-text"
        >
          Next
        </span>
        <span
          aria-hidden={true}
          className="fa fa-angle-right"
          id="Icon1"
        />
      </div>
    </a>
  </div>
</div>
`;
